scripts/Microsoft.DotNet.Cli.Build.Framework/BuildHelpers.cs

```diff
@@ -5,38 +5,47 @@ namespace Microsoft.DotNet.Cli.Build.Framework
-        public static int ExecInSilent(string workingDirectory, string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory, silent: true);
+        public static int ExecInSilent(string workingDirectory, string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory, silent: true, env: null);
 
         public static int ExecIn(string workingDirectory, string command, params string[] args) => ExecIn(workingDirectory, command, (IEnumerable<string>)args);
-        public static int ExecIn(string workingDirectory, string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory, silent: false);
+        public static int ExecIn(string workingDirectory, string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory, silent: false, env: null);
 
         public static int ExecSilent(string command, params string[] args) => ExecSilent(command, (IEnumerable<string>)args);
-        public static int ExecSilent(string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory: null, silent: true);
+        public static int ExecSilent(string command, IEnumerable<string> args) => ExecSilent(command, args, env: null);
+        public static int ExecSilent(string command, IEnumerable<string> args, IDictionary<string, string> env) => ExecCore(command, args, workingDirectory: null, silent: true, env: null);
 
         public static int Exec(string command, params string[] args) => Exec(command, (IEnumerable<string>)args);
-        public static int Exec(string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory: null, silent: false);
+        public static int Exec(string command, IEnumerable<string> args) => ExecCore(command, args, workingDirectory: null, silent: false, env: null);
 
         public static Command Cmd(string command, params string[] args) => Cmd(command, (IEnumerable<string>)args);
         public static Command Cmd(string command, IEnumerable<string> args)
         {
             return Command.Create(command, args);
         }
 
-        internal static int ExecCore(string command, IEnumerable<string> args, string workingDirectory, bool silent)
+        internal static int ExecCore(string command, IEnumerable<string> args, string workingDirectory, bool silent, IDictionary<string, string> env)
         {
             var cmd = Cmd(command, args);
-            if(!string.IsNullOrEmpty(workingDirectory))
+            if (!string.IsNullOrEmpty(workingDirectory))
             {
                 cmd.WorkingDirectory(workingDirectory);
             }
-            if(silent)
+
+            if (silent)
             {
                 cmd.CaptureStdErr().CaptureStdOut();
             }
-            var result = cmd.Execute();
+
+            if (env == null)
+            {
+                env = new Dictionary<string, string>();
+            }
+
+            var result = cmd.Environment(env).Execute();
```

Comment:
Maybe cmd.Environment should allow passing in `null` and just no-oping?  That way we don't waste creating an empty dictionary for no reason.